assignment3.txt

```diff
@@ -0,0 +1,71 @@
+Jinoo Hong
+
+1.  Determine the complexity of the following code snippets: 
+
+    a.  
+    ``````C
+      int sum = 0;
+      for(i = 0; i < n; i++)
+              sum++;
+    ```
+O(n)
+    
+    b.  
+    ``` C
+    sum = 0;
+    for( i = 0; i < n; i++ )
+        sum++;
+    for( i = 0; i < n; i++ )
+        for( j = 0; j < n; j++ )
+            sum++;
+    ```
+O(2n)
```

Comment:
Not quite! you should have given an explanation so I can tell you where you went wrong. the time complexity of this is n^2 + n, which evaluates to O(n^2). The n^2 comes from the nested for loops in the second half of the code snippet and the n comes from the first for loop.